pvmismatch/contrib/ModuleMismatchSimulator.py

```diff
@@ -0,0 +1,512 @@
+                      'sy': shd.sy, 'sx': shd.sx,
+                      'numberCells': pvmod1.numberCells})
+    filename = raw_input("File name? ")
+    filename = 'JSONshade/' + filename + '.json'
+    fo = open(filename, 'w')
+    shades_json = json.dump(dicts, fo, sort_keys=True, indent=2)
+    fo.close()
+
+
+def clear_last_full(val, update=None):
+    if len(ivp0.shade) != 1:
+        ivp0.shade.pop(-2)
+    update(val)
+
+
+if __name__ == "__main__":
+    pvsys, modHeight, numcells = pvsys_defs_user_input(npts=201)
+    output = construct_plot_area(numcells, modHeight)
+    pvmod1 = pvsys.pvstrs[0].pvmods[0]
+    pvmod_noshade = deepcopy(pvmod1)
+    calcmod_noshade = pvmod_noshade.calcMod()
+    Pmod_noshade = calcmod_noshade[2].squeeze()
+    Pmp0 = Pmod_noshade.max()
+    ivp0 = IVP()
+    plotobjs = PlotObjs()
+    update = partial(full_update, output=output, ivp0=ivp0, plotobjs=plotobjs)
+    ClearLast = partial(clear_last_full, update=update)
+    print "Pmp0: {}".format(Pmp0)
+    print ""
+    print '{0:6} {1:^6} {2:^6} {3:^7}'.format('#Cells', '%Shade', 'Pmp', '%ofPmp0')
+    print '----------------------------'
+    ps0 = 90
+    sw0 = 1
+    sh0 = 1
+    sy0 = 1
+    sx0 = 1
+    ivp0.shade = []
+    all_calc(ivp0, ps0, sw0, sh0, sx0, sy0)
+    plot_init(ivp0, plotobjs, output['ax00'], output['ax01'], output['ax10'], output['ax11'], output['ax02'],
+              output['ax12'], output['ax03'], output['ax_4'], output['x'], output['y'])
+    plt.draw()
+
+    cellPlot = output['cellPlot']
+    ax_ps = cellPlot.add_axes([0.64, 0.38, 0.12, .04])
+    ax_sw = cellPlot.add_axes([0.64, 0.33, 0.12, .04])
+    ax_sh = cellPlot.add_axes([0.64, 0.28, 0.12, .04])
+    ax_sy = cellPlot.add_axes([0.64, 0.23, 0.12, .04])
+    ax_sx = cellPlot.add_axes([0.64, 0.18, 0.12, .04])
+    ax_button_setshade = cellPlot.add_axes([0.63, 0.10, 0.06, 0.06])
+    ax_button_saveshade = cellPlot.add_axes([0.71, 0.10, 0.06, 0.06])
+    ax_button_clearlast = cellPlot.add_axes([0.67, 0.03, 0.06, 0.06])
+    s_ps = Slider(ax_ps, 'Shade%', 0, 100, valinit=ps0)
+    s_sw = Slider(ax_sw, 'Width', 1, numcells / modHeight, valinit=sw0,
+                  valfmt='%0.0f')
+    s_sh = Slider(ax_sh, 'Height', 1, modHeight, valinit=sh0, valfmt='%0.0f')
+    s_sy = Slider(ax_sy, 'Row Start', 1, modHeight, valinit=sw0,
+                  valfmt='%0.0f')
+    s_sx = Slider(ax_sx, 'Col Start', 1, numcells / modHeight, valinit=sh0,
+                  valfmt='%0.0f')
+    b_setshade = Button(ax_button_setshade, "Set Shade")
+    b_saveshade = Button(ax_button_saveshade, "Save Shade")
+    b_clearlast = Button(ax_button_clearlast, "Clear Last")
+    s_ps.on_changed(update)
+    s_sw.on_changed(update)
+    s_sh.on_changed(update)
+    s_sy.on_changed(update)
+    s_sx.on_changed(update)
+    b_setshade.on_clicked(set_the_shade)
+    b_saveshade.on_clicked(save_the_shade)
+    b_clearlast.on_clicked(ClearLast)
+
+plt.show()
```

Comment:
also please change the name of the module to conform to pep8:
>Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged.

Something all lower case like `simulator.py` or `mismatch-simulator.py` - since it's a script it's okay to have a dash in it, just means you can't import it from another module, which you'll never need to do anyway, unless you want to in which case, do **not** put a dash in the name, use underscore instead like `module_mismatch_simulator.py`. Whatever you think is best, just not camel case, save that for classes.